From 8b844e344b8417aba6877ae8b78f393ed86b4a0f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jakab=20Krist=C3=B3f?= <jakab.kristof@balasys.hu>
Date: Thu, 14 Jul 2016 08:37:37 +0200
Subject: [PATCH 07/19] Revert "UBUNTU: SAUCE: UEFI: efi: Disable secure boot
 if shim is in insecure mode"

This reverts commit 1b1045ddaeaa42a98de2fa0039e4ae885e189cec.
---
 arch/x86/boot/compressed/eboot.c | 20 +-------------------
 1 file changed, 1 insertion(+), 19 deletions(-)

diff --git a/arch/x86/boot/compressed/eboot.c b/arch/x86/boot/compressed/eboot.c
index e5c7deb..1b13869 100644
--- a/arch/x86/boot/compressed/eboot.c
+++ b/arch/x86/boot/compressed/eboot.c
@@ -834,9 +834,8 @@ out:
 
 static int get_secure_boot(void)
 {
-	u8 sb, setup, moksbstate;
+	u8 sb, setup;
 	unsigned long datasize = sizeof(sb);
-	u32 attr;
 	efi_guid_t var_guid = EFI_GLOBAL_VARIABLE_GUID;
 	efi_status_t status;
 
@@ -860,23 +859,6 @@ static int get_secure_boot(void)
 	if (setup == 1)
 		return 0;
 
-	/* See if a user has put shim into insecure_mode.  If so, and the variable
-	 * doesn't have the runtime attribute set, we might as well honor that.
-	 */
-	var_guid = EFI_SHIM_LOCK_GUID;
-	status = efi_early->call((unsigned long)sys_table->runtime->get_variable,
-				L"MokSBState", &var_guid, &attr, &datasize,
-				&moksbstate);
-
-	/* If it fails, we don't care why.  Default to secure */
-	if (status != EFI_SUCCESS)
-		return 1;
-
-	if (!(attr & EFI_VARIABLE_RUNTIME_ACCESS)) {
-		if (moksbstate == 1)
-			return 0;
-	}
-
 	return 1;
 }
 
-- 
2.9.2

